File: unstructured/sphere.rb
# frozen_string_literal: true

Point = Struct.new(:x, :y, :z)
filename = "sphere.vtk"
f = open(filename, "w")
puts filename

vp = Array.new(10000) do
  z = rand*2-1.0
  s = rand*2.0*Math::PI
  x = (1-z**2)**0.5*Math.cos(s)
  y = (1-z**2)**0.5*Math.sin(s)
  Point.new(x, y, z)
end

f.puts "# vtk DataFile Version 2.0"
f.puts "unstructured"
f.puts "ASCII"
f.puts "DATASET UNSTRUCTURED_GRID"
f.puts "POINTS #{vp.size} float"
vp.each do |v|
  f.puts "#{v.x} #{v.y} #{v.z}"
end

f.puts "POINT_DATA #{vp.size}"
f.puts "VECTORS vector float"
vp.each do |v|
  f.puts "#{v.y} #{-v.x} 0"
end

f.puts "SCALARS z float"
f.puts "LOOKUP_TABLE defalut"
vp.each do |v|
  f.puts v.z.to_s
end

File: unstructured/sphere.py
from math import pi, cos, sin
from random import random


class Point:
    def __init__(self, x, y, z):
        self.x = x
        self.y = y
        self.z = z


filename = "sphere.vtk"
f = open(filename, "w")
print(filename)

vp = []
for i in range(10000):
    z = random() * 2.0 - 1.0
    s = random() * 2.0 * pi
    x = (1 - z**2)**0.5 * cos(s)
    y = (1 - z**2)**0.5 * sin(s)
    vp.append(Point(x, y, z))

f.write(f"""\
# vtk DataFile Version 2.0
unstructured
ASCII
DATASET UNSTRUCTURED_GRID
POINTS {len(vp)} float
""")

for v in vp:
    f.write(f"{v.x} {v.y} {v.z}\n")

f.write(f"""\
POINT_DATA {len(vp)}
VECTORS vector float
""")

for v in vp:
    f.write(f"{v.y} {-v.x} 0\n")

f.write("SCALARS z float\n")
f.write("LOOKUP_TABLE defalut\n")
for v in vp:
    f.write(f"{v.z}\n")
